src/app/campaign.service.spec.ts

```diff
@@ -98,6 +105,23 @@ describe('CampaignService', () => {
     request.flush(dummyCampaign);
   });
 
+  it('should retrieve campaign totals from API', () => {
+    const service: CampaignService = TestBed.inject(CampaignService);
+    const httpMock: HttpTestingController = TestBed.inject(HttpTestingController);
+
+    const dummyCampaignStats: CampaignStats = getDummyCampaignStats();
+
+    service.getCampaignImpactStats().subscribe(campaignStats => {
+      expect(campaignStats).toEqual(dummyCampaignStats);
+    }, () => {
+      expect(false).toBe(true);
+    });
+
+    const request = httpMock.expectOne(`${environment.apiUriPrefix}/campaigns/services/apexrest/v1.0/campaigns/stats`);
+    expect(request.request.method).toBe('GET');
+    request.flush(dummyCampaignStats);
+  });
```

Comment:
the Subject Under Test of this code is just the one liner function `getCampaignImpactStats`, right? That function seems so simple I'm not convinced it's worth testing, especially with a test this long. What sort of mistakes would we expect this test to catch / prevent that the compiler won't catch automatically?

Also the last three lines don't seem to be anything to do with the SUT - what are they for?

Reply:
On the question of whether to test every function - this is a good answer from Kent Beck, author of Test Driven Development:

https://stackoverflow.com/questions/153234/how-deep-are-your-unit-tests/153565#153565